lib/facter/simplib_networkmanager.rb

```diff
@@ -0,0 +1,49 @@
+Facter.add(:simplib_networkmanager) do
+  confine :kernel => 'Linux'
+
+  @nmcli_cmd = Facter::Util::Resolution.which('nmcli')
+  confine { @nmcli_cmd }
+
+  setcode do
+    info = { 'enabled' => false }
+
+    nmcli_cmd = @nmcli_cmd + ' -t'
+
+    general_status = Facter::Core::Execution.execute(%(#{nmcli_cmd} -m multiline general status), :on_fail => :failed)
+    unless general_status == :failed
+      general_status = general_status.lines.map{|line| line.strip.split(':') }
+
+      info['enabled'] = true
+      info = {
+        'general' => {
+          'status' => Hash[general_status]
+        }
+      }
+    end
+
+    general_hostname = Facter::Core::Execution.execute(%{#{nmcli_cmd} general hostname}, :on_fail => :failed)
+    unless general_hostname == :failed
+      info['enabled'] = true
+      info['general'] ||= {}
+      info['general']['hostname'] = general_hostname.strip
+    end
+
+    connections = Facter::Core::Execution.execute(%(#{nmcli_cmd} connection show), :on_fail => :failed)
+    unless connections == :failed
+      info['enabled'] = true
+      info['connection'] = {}
+
+      connections.lines.each do |conn|
+        name, uuid, type, device = conn.strip.split(':')
```

Comment:
Reading @lnemsick-simp's question as "would something else make a more useful key?":  

The `id` is human-readable, but not optimal for manifest logic.  It's more likely to change than the (connection) `uuid`, and neither `id` or `uuid` describe anything any other Puppet facts refer to—such the device, which seems like the obvious key someone would want to correlate with this fact. 

On the face of it, it seems like we should either:
* use the device (`br1`) instead of the id the key, or 
* add an `info['device']` subtree with the output of `nmcli -t device`

Either solution would simplify the process of looking up connection information from a device name.

Reply:
@op-ct Unfortunately, if you want to actually use some of the `nmcli` commands, you need to use the name instead of the device from what I can tell.

Reply:
@op-ct I *think* that you can also have multiple entries pointing to the same device that are in different states of activation which is why you need the `name` as the key.

Reply:
@trevor-vaughan 

> @op-ct Unfortunately, if you want to actually use some of the `nmcli` commands, you need to use the name instead of the device from what I can tell.

That is precisely why either of the solutions above would result in a more useful Puppet fact:  

* No other Puppet facts or parameters report the NetworkManager connection `id` (or the interchangeable but similarly oblique `uuid` or (DBUS) `path`)—but they always report  a system's `interfaces`.   Puppet code trying to use this fact will almost certainly start with the interface (device) and need to map it to the connection in order to look up which one it needs.
  * If we used the device (`br1`) instead of the `id` as the key in the fact, or added an `info['device']` subtree to the fact structure (populated with the output of `nmcli -t device`), we could start with the device and just look it up.
  * Alternatively, adding an `info['device']` subtree to the fact structure (populated with the output of `nmcli -t device`) is still just two key lookups, which is more useful than coming up with your own data structure `dig()` logic in each place you reference the fact.
* Additionally, `id` does not make a very trustworthy identifier.  It is designed to be changed at any time (`nmcli con modify STRING1 connection.id STRING2`), unlike the device name.



> @op-ct I _think_ that you can also have multiple entries pointing to the same device that are in different states of activation which is why you need the `name` as the key.

According to the [`nmcli` man page](https://developer.gnome.org/NetworkManager/stable/nmcli.html), there can be multiple connections that apply to a device—but only one of them can be active on that device at any given time.